### ISSUE TITLE
Applying negative Sliding Stretch on a selection less than 0.2s freezes audacity

### ISSUE THREAD

**Issue by aaronliu0130** [audacity/audacity#5793]
### Steps to reproduce

1. Import an audio track or sth, just get audio
2. At the end of the audio select a part of it that lasts less than 0.2s
3. Try to apply a sliding stretch

### Expected behavior

I get a warning that you can't do this or it applies the sliding stretch

### Actual behavior

Audacity freezes

### Audacity Version

latest stable version (from audacityteam.org/download)

### Operating system

Windows, macOS, & endeavourOS

**Comment by petersampsonaudacity** [comment]
Testing on W10 with 3.4.2

I cannot reproduce this - I tested with both mono and stereo ( a 30 second chirp) - I tried just under 0.2 secs at the end of the chirp and 0.01 sec at the end of the chirp.  Both worked in mono and stereo.

**Comment by dozzzzer** [comment]
Couldn't reproduce this either (macOS 14.2/Windows 11)

**Comment by aaronliu0130** [comment]
That is weird. My Audacity is at 3.4.2, the latest... Is there something I could do to diagnose this?

**Comment by dozzzzer** [comment]
@aaronliu0130 yes, you might want to share your project with us via Dropbox/Google Drive/etc which we will look into.

**Comment by aaronliu0130** [comment]
For me it works on any audio file I open, so I don’t think that’ll be useful.

**Comment by dozzzzer** [comment]
> For me it works on any audio file I open

Well, I've tried opening quite a few mono/stereo audio files and applying Sliding Stretch to them but none of them showed the issue you had experienced.

**Comment by aaronliu0130** [comment]
Is there some kind of logs I could get?

**Comment by dozzzzer** [comment]
> Is there some kind of logs I could get?

You may want to look into Help > Diagnostics > Show Log

**Comment by aaronliu0130** [comment]
The logs appear to reset every startup.

**Comment by aaronliu0130** [comment]
Is there a way I can debug my problem?

**Comment by aaronliu0130** [comment]
@dozzzzer Can you reproduce if the stretch's final tempo change is negative? Positive seems to also work for me, but not negative.

**Comment by aaronliu0130** [comment]
@LWinterberg Hopefully you don't mind the ping! But can you confirm that this is reproducible?

**Comment by aaronliu0130** [comment]
@dozzzzer Can you reproduce if the stretch's final tempo change is negative?

**Comment by dozzzzer** [comment]
@aaronliu0130 of course, there you go

https://github.com/audacity/audacity/assets/70063970/888c35fa-07b7-4b4c-85e0-ad25b83bb2be

**Comment by aaronliu0130** [comment]
@dozzzzer Thanks. The value after which the sliding stretch freezes seems to vary between different samples for me as well. Would you kindly try making the final tempo change -90? That seems to consistently freeze for me.

**Comment by aaronliu0130** [comment]
@dozzzzer I've successfully reproduced this on endeavourOS as well. -90% consistently reproduces things.

**Comment by dozzzzer** [comment]
@aaronliu0130 with -90% I've finally managed to reproduce the bug
